pkg/resources/database_role_acceptance_test.go

```diff
@@ -0,0 +1,118 @@
+package resources_test
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+	"text/template"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+type (
+	TestAccDatabaseRoleSettings struct {
```

Comment:
i think this is a bit overkill. probably better to just have a function like `databaseRoleConfig(warehouseName, databaseName, roleName`)